CRM/Contact/Form/Search.php

```diff
@@ -302,7 +302,8 @@ public static function getModeValue($mode = CRM_Contact_BAO_Query::MODE_CONTACTS
     }
 
     self::setModeValues();
-    if (!array_key_exists($mode, self::$_modeValues)) {
```

Comment:
or we could cast to an array 

if (!array_key_exists($mode, (array) self::$_modeValues)) {

Reply:
Would that help because the warning is the first param, because of the array_search above $mode sometimes ends up being FALSE.

Reply:
ah